scripts/kind-with-registry.sh

```diff
@@ -17,48 +17,72 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# Install kubectl
+# Install kubectl and kind
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 KUBECTL="${REPO_ROOT}/hack/tools/bin/kubectl"
 KIND="${REPO_ROOT}/hack/tools/bin/kind"
 make --directory="${REPO_ROOT}" "${KUBECTL##*/}" "${KIND##*/}"
 
-# desired cluster name; default is "kind"
+# Export desired cluster name; default is "capz"
 KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-capz}"
 export KIND_CLUSTER_NAME
 
-if [[ "$("${KIND}" get clusters)" =~ .*"${KIND_CLUSTER_NAME}".* ]]; then
+if [[ "$("${KIND}" get clusters)" = "${KIND_CLUSTER_NAME}" ]]; then
```

Comment:
Some of the times, I have seen kind clusters having a prefix `kind` to the cluster names. (I must have missed the kind configuration when that happens tho.)

Do you think dropping the regular expression check here is (`.*"${KIND_CLUSTER_NAME}".*`) is beneficial ?

Reply:
Oh good catch @nawazkh! This regex check removal was actually a change I made locally when testing that I intended to revert but hadn't. Fixed now.